bpython/autocomplete.py

```diff
@@ -265,27 +286,31 @@ def shown_before_tab(self) -> bool:
 class CumulativeCompleter(BaseCompletionType):
     """Returns combined matches from several completers"""
 
-    def __init__(self, completers, mode=AutocompleteModes.SIMPLE) -> None:
+    def __init__(
+        self,
+        completers: Sequence[BaseCompletionType],
+        mode: AutocompleteModes = AutocompleteModes.SIMPLE,
+    ) -> None:
         if not completers:
             raise ValueError(
                 "CumulativeCompleter requires at least one completer"
             )
-        self._completers = completers
+        self._completers: Sequence[BaseCompletionType] = completers
 
         super().__init__(True, mode)
 
-    def locate(self, current_offset: int, line: str) -> Union[None, NoReturn]:
+    def locate(self, cursor_offset: int, line: str) -> Union[LinePart, None]:
```

Comment:
`Optional[LinePart]` would work for the return type too.

Reply:
I got rid of all of these except a few
```
-> Union[str, curtsies.events.Event, None]:
```
in curtsies.py.